carsus/io/cmfgen/util.py

```diff
@@ -0,0 +1,332 @@
+
+        energy_div_threshold = 1 + 20 * (c ** 2)
+        u = energy_div_threshold
+        x = np.log10(min(u, e))
+
+        cross_section = 10 ** (a + x * (b + x * (c + x * d)))
+        if u > e:
+            cross_section *= (e / u) ** 2
+
+        phixs_table[i] = energy_div_threshold * threshold_energy_ryd, cross_section
+
+    return phixs_table
+
+
+def get_hummer_phixs_table(threshold_energy_ryd, a, b, c, d, e, f, g, h, n_points=1000):
+    """
+    References:
+
+        A Fast and Accurate Method for Evaluating the Nonrelativistic Free-free
+        Gaunt Factor for Hydrogenic Ions.
+        
+        Hummer, D. G.
+
+        Astrophysical Journal v.327, p.477
+    """
+
+    # Only applies to `He`. The threshold cross sections seems ok, but 
+    # energy dependence could be slightly wrong. What is the `h` parameter
+    # that is not used?.
+    
+    energy_grid = np.linspace(0.0, 1.0, n_points, endpoint=False)
+    phixs_table = np.empty((len(energy_grid), 2))
+
+    for i, c in enumerate(energy_grid):
+        energy_div_threshold = 1 + 20 * (c ** 2)
+
+        x = np.log10(energy_div_threshold)
+        if x < e:
+            cross_section = 10 ** (((d * x + c) * x + b) * x + a)
+
+        else:
+            cross_section = 10 ** (f + g * x)
+
+        phixs_table[i] = energy_div_threshold * threshold_energy_ryd, cross_section
+
+    return phixs_table
+
+
+def get_vy95_phixs_table(threshold_energy_ryd, fit_coeff_table, n_points=1000):
+    """
+    References:
+
+        Analytic FITS for partial photoionization cross sections.
+
+        Verner, D. A. ; Yakovlev, D. G.
+
+        Astronomy and Astrophysics Suppl., Vol. 109, p.125-133 (1995)
+    """
+    energy_grid = np.linspace(0.0, 1.0, n_points, endpoint=False)
+    phixs_table = np.empty((len(energy_grid), 2))
+
+    for i, c in enumerate(energy_grid):
+
+        energy_div_threshold = 1 + 20 * (c ** 2)
+        cross_section = 0.0
+
+        for index, row in fit_coeff_table.iterrows():
+            y = energy_div_threshold * row.at['E'] / row.at['E_0']
+            P = row.at['P']
+            Q = 5.5 + row.at['l'] - 0.5 * row.at['P']
+            y_a = row.at['y(a)']
+            y_w = row.at['y(w)']
+            cross_section += row.at['sigma_0'] * ((y - 1) ** 2 + y_w ** 2) * (y ** -Q) * ((1 + np.sqrt(y / y_a)) ** -P)
+
+        phixs_table[i] = energy_div_threshold * threshold_energy_ryd, cross_section
+
+    return phixs_table
+
+
+def get_leibowitz_phixs_table(threshold_energy_ryd, a, b, c, d, e, f, n_points=1000):
+    """
+    References:
+
+        Radiative Transition Probabilities and Recombination Coefficients 
+        of the Ion C IV.
+
+        Leibowitz, E. M.
+
+        J. Quant. Spectrosc. Radiat. Transfer. Vol 12, pp. 299-306.
+    """
+    
+    raise NotImplementedError
+
+
+def get_null_phixs_table(n_points=1000):
+    """
+    """
+    phixs_table = np.column_stack((np.linspace(0, 1.0, 1000, endpoint=False), 
```

Comment:
This is not particularly crucial but it would be nice if the energies start at the threshold energy (as for the other cross-section types) and not at 0.